Tools/Scripts/libraries/webkitcorepy/webkitcorepy/tests/task_pool_unittest.py

```diff
@@ -170,3 +173,75 @@ def test_teardown_arguments(self):
                 sorted(captured.webkitcorepy.log.getvalue().splitlines()),
                 ['worker/{} Teardown argument'.format(x) for x in range(4)],
             )
+
+        def test_mutually_exclusive_group(self):
+            with OutputCapture(level=logging.INFO) as captured:
+                with TaskPool(workers=4, mutually_exclusive_groups=dict(group=lambda c, **kwargs: c in ['a', 'b', 'c', 'd'])) as pool:
```

Comment:
As I understand it (and I only really understand it from the tests), `mutually_exclusive_groups` is a dict from group name (`str`) to a callable that is truthful if it is a member of that group? And the callable gets passed the arguments the action receives?